caste.go

```diff
@@ -612,14 +696,29 @@ func ToUint8E(i interface{}) (uint8, error) {
 		}
 		return 0, fmt.Errorf("unable to cast %#v to uint8: %s", i, err)
 	case int:
+		if s < 0 {
+			return 0, fmt.Errorf("unable to cast negative value to uint8")
```

Comment:
You are over-engineering slightly. People know what func they called, so you could just add a private member err variable at the top of the file with a more generic message, `errNegativeNotAllowed = "unable to cast negative value"`, and then return that.